lxd/auth/generate/main.go

```diff
@@ -0,0 +1,256 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"sort"
+	"strings"
+	"unicode"
+
+	"github.com/canonical/lxd/shared/entity"
+	"github.com/canonical/lxd/shared/logger"
+)
+
+var typeRegexp = regexp.MustCompile(`^type (\w+)$`)
+var relationRegexp = regexp.MustCompile(`^\s+define\s+(\w+):\s+.+$`)
+var commentRegexp = regexp.MustCompile(`^\s*#\s*(.*)$`)
+
+type entitlement struct {
+	relation    string
+	description string
+}
+
+func snakeToPascal(str string) string {
+	// Capitalise first letter.
+	capitalise := true
+	var b strings.Builder
+	for _, r := range str {
+		if capitalise {
+			b.WriteRune(unicode.ToUpper(r))
+			capitalise = false
+			continue
+		}
+
+		// Capitalise letters that occur after underscores (and skip the underscores).
+		if r == '_' {
+			capitalise = true
+			continue
+		}
+
+		b.WriteRune(r)
+	}
+
+	// Map of expected incorrect pascal case acronyms to the correct casing.
+	knownAcronyms := map[string]string{
+		"Acl":  "ACL",
+		"Sftp": "SFTP",
+	}
+
+	s := b.String()
+	for wrong, right := range knownAcronyms {
+		s = strings.Replace(s, wrong, right, -1)
+	}
+
+	return s
+}
+
+func main() {
+	f, err := os.Open("driver_openfga_model.openfga")
```

Comment:
If you moved all of these function calls in main into a single anonymous function that returns an error you could then have a single `fmt.Printf() os.Exit(1)` handler at the end if it returned error, rather than having to handle it repeatedly and increase the chance of introducing a bug.

Reply:
we should probably have a defer f.Close() here. I realise you're handling it by os.Exit, but in case it gets refactored in the future, it wont hurt to call it twice, once explicitly on success and once by the defer.